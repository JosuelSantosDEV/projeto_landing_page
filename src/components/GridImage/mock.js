
const srcImg = "https://plus.unsplash.com/premium_photo-1673909405731-94b05ba69308?fm=jpg&q=60&w=3000&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8c291cmNlfGVufDB8fDB8fHww"
export default {
    background: false,
    title: "My grid",
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. 
    Accusantium commodi, vero quis
     atque architecto minus debitis nobis
     harum amet, illo, saepe enim velit eum a eveniet id aut totam quas.`,
     grid: [
        {
            altText: "Title 1",
            srcImg
        },
        {
            altText: "Title 2",
            srcImg
        },
        {
            altText: "Title 3",
            srcImg
        },
        {
            altText: "Title 4",
            srcImg
        },
        {
            altText: "Title 5",
            srcImg
        },
        {
            altText: "Title 6",
            srcImg
        },
     ]
}