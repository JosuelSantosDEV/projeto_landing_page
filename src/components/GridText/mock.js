export default {
    background: false,
    title: "My grid",
    description: "You’ve probably heard of Lorem Ipsum before – it’s the most-use",
    grid: [
        {
            title: "Test 1",
            description: `You’ve probably heard of Lorem Ipsum before – it’s the most-used dummy 
            text excerpt out there. People use it because it has a fairly normal distribution of 
            letters and words (making it look like normal English), but it’s also Latin, which 
            means your average reader won’t get distracted by trying to read it. It’s perfect for .`
        },
        {
            title: "Test 2",
            description: `You’ve probably heard of Lorem Ipsum before – it’s the most-used dummy 
            text excerpt out there. People use it because it has a fairly normal distribution of 
            letters and words (making it look like normal English), but it’s also Latin, which 
            means your average reader won’t get distracted by trying to read it. It’s perfect for .`
        },
        {
            title: "Test 3",
            description: `You’ve probably heard of Lorem Ipsum before – it’s the most-used dummy 
            text excerpt out there. People use it because it has a fairly normal distribution of 
            letters and words (making it look like normal English), but it’s also Latin, which 
            means your average reader won’t get distracted by trying to read it. It’s perfect for .`
        }
    ]
}