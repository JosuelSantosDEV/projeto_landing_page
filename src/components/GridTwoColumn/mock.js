export default {
    title:"Essa Ã© a GridTwoColumn",
    text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. 
    Accusantium commodi, vero quis
     atque architecto minus debitis nobis
     harum amet, illo, saepe enim velit eum a eveniet id aut totam quas.`,
    srcImg: "assets/images/javascript.svg",
    background: false
}