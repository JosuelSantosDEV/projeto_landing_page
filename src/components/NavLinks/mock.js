export default [
    {
        children: "Home",
        link: "/home",
        newTab: true
    },
    {
        children: "About",
        link: "/about",
        newTab: true
    },
    {
        children: "Favorite",
        link: "/favorite",
        newTab: true
    }
]