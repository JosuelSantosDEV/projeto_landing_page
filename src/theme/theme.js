export const theme = {
    colors: {
        dark: "#000",
        white: "#fff",
        gray: "#c1c1c1"
    },
    fonts: {

    }
}