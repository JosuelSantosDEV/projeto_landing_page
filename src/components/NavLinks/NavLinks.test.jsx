import { renderTheme } from "../../theme/render-theme";
import { screen } from "@testing-library/react";
import { NavLinks } from ".";

import mock from "./mock";
import { theme } from "../../theme/theme";

describe("<NavLinks />", () => {

    it("Should render links", () => {
        renderTheme(<NavLinks links={mock}/>);
        const tag = screen.getAllByRole("link");
        expect(tag).toHaveLength(mock.length);
    }
    );

    it("Should not render link", () => {
        renderTheme(<NavLinks />);
        const tag = screen.queryAllByRole("link");
        expect(tag).toHaveLength(0);
    }
    );

    it("Should not render link", () => {
        const {debug} = renderTheme(<NavLinks />);
        //debug()
        const tag = screen.getByRole("navigation");
        expect(tag).toHaveStyleRule('flex-flow',"column wrap",{
            media: theme.media.lteMedium
        });
    }
    );


    it("Should match snapshot", () => {
        const {container} = renderTheme(<NavLinks links={mock}/>);
        expect(container.firstChild).toMatchSnapshot();
    }
    );


});