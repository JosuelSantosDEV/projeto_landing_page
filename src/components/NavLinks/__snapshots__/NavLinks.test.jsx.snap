// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="/home"
    target="_blank"
  >
    Home
  </a>
  <a
    class="c1"
    href="/about"
    target="_blank"
  >
    About
  </a>
  <a
    class="c1"
    href="/favorite"
    target="_blank"
  >
    Favorite
  </a>
</nav>
`;
